lib/sdr_client/redesigned_client/job_status.rb

```diff
@@ -35,11 +37,26 @@ def errors
       # @param [Integer] timeout_in_secs (180) timeout after this many secs
       # @param [Float] backoff_factor (2.0) how quickly to backoff. This should be > 1.0 and probably ought to be <= 2.0
       # @return [Boolean] true if successful false if unsuccessful.
-      def wait_until_complete(secs_between_requests: 3.0,
+      def wait_until_complete(secs_between_requests: 3.0, # rubocop:disable Metrics/MethodLength
                               timeout_in_secs: 180,
                               backoff_factor: 2.0,
                               max_secs_between_requests: 60)
-        poll_until_complete(secs_between_requests, timeout_in_secs, backoff_factor, max_secs_between_requests)
+        begin
+          Timeout.timeout(timeout_in_secs) do
+            loop do
+              break if complete?
+
+              yield if block_given?
```

Comment:
The only consumer that sends a block is the CLI, which prints dots to show the user that the polling is in progress.